packages/react-tooltip-trigger/src/components/TooltipTrigger/useTooltipTrigger.ts

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useId } from '@fluentui/react-utilities';
+import { TooltipTriggerChildProps, TooltipTriggerProps, TooltipTriggerState } from './TooltipTrigger.types';
+import { TooltipImperativeHandle } from '../../common/TooltipProps.types';
+import { TooltipTriggerReason } from '../../common/TooltipManager.types';
+import { useTooltipContext } from '../../common/useTooltipContext';
+
+/**
+ * Names of the shorthand properties in TooltipTriggerProps
+ * {@docCategory TooltipTrigger}
+ */
+export const tooltipTriggerShorthandProps = ['tooltip'] as const;
+
+const mergeProps = makeMergeProps<TooltipTriggerState>({ deepMerge: tooltipTriggerShorthandProps });
+
+/**
+ * Create the state required to render TooltipTrigger.
+ *
+ * The returned state can be modified with hooks such as useTooltipTriggerStyles,
+ * before being passed to renderTooltipTrigger.
+ *
+ * @param props - props from this instance of TooltipTrigger
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ * {@docCategory TooltipTrigger}
+ */
+export const useTooltipTrigger = (
+  props: TooltipTriggerProps,
+  defaultProps?: TooltipTriggerProps,
+): TooltipTriggerState => {
+  const tooltipRef = React.useRef<TooltipImperativeHandle | null>(null);
+
+  const { Tooltip, manager, portalRoot } = useTooltipContext();
+
+  const state = mergeProps(
+    {
+      showDelay: 250,
+      hideDelay: 250,
+      children: props.children,
+      tooltip: {
+        as: Tooltip,
+        id: useId('tooltip-'),
+        componentRef: tooltipRef,
+        onPointerEnter: ev => manager?.onPointerEnterTooltip(ev.currentTarget),
+        onPointerLeave: ev => manager?.onPointerLeaveTooltip(ev.currentTarget),
```

Comment:
`onPointerEnter` & `onPointerLeave` are DOM handlers correct? In this case we should use overrides&merge them as otherwise user's input will win:

```ts
<TooltipTrigger tooltip={{ componentRef /* wins over ours */, onPointerEnter /* wins over ours */ }} />
```